apps/proxy/src/delete-cache.ts

```diff
@@ -0,0 +1,105 @@
+import * as fs from 'node:fs/promises'
+import * as path from 'node:path'
+import { exec } from 'node:child_process'
+import { promisify } from 'node:util'
+import { env } from './env.js'
+const execAsync = promisify(exec)
+
+async function deleteOldFolders() {
+  const now = Date.now()
+  const ttlInMillis = env.CACHE_TTL * 60 * 60 * 1000
+
+  try {
+    const folders = await fs.readdir(env.CACHE_PATH)
+    for (const folder of folders) {
+      const folderPath = path.join(env.CACHE_PATH, folder)
+      const stats = await fs.stat(folderPath)
+
+      if (stats.isDirectory() && now - stats.mtimeMs > ttlInMillis) {
+        await fs.rm(folderPath, { recursive: true, force: true })
+        console.log(`Deleted folder: ${folderPath}`)
+      }
+    }
+  } catch (err) {
+    console.error('Failed to delete old folders:', err)
+  }
+}
+
+async function scriptAlreadyRan() {
+  try {
+    const lastRun = parseInt(await fs.readFile(env.CACHE_TIMESTAMP_FILE, 'utf8'))
+    const now = Math.floor(Date.now() / 1000)
+    const diff = now - lastRun
+    return diff < env.CACHE_SCHEDULE_INTERVAL * 60 * 60 * 1000
+  } catch (err) {
+    // File does not exist
+    if (err instanceof Error && 'code' in err && err.code === 'ENOENT') {
+      return false
+    }
+    throw err
+  }
+}
+
+async function updateTimestampFile() {
+  const now = Math.floor(Date.now() / 1000).toString()
+  await fs.writeFile(env.CACHE_TIMESTAMP_FILE, now)
+}
+
+async function getDiskUsage() {
+  const command = `df / | awk 'NR==2 {print $5}' | sed 's/%//'`
```

Comment:
Can we add a comment with some details on this command? It's not clear exactly how it works - is it targeting a specific FS row/column?

Reply:
Yes, I added a comment.